content/docs/2.19/scalers/kubernetes-resource.md

```diff
@@ -0,0 +1,273 @@
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: app-metrics
+  namespace: keda
+data:
+  stats: |
+    application:
+      workers:
+        active: 15
+        idle: 3
+---
+apiVersion: keda.sh/v1alpha1
+kind: ScaledObject
+metadata:
+  name: worker-scaler
+  namespace: keda
+spec:
+  scaleTargetRef:
+    name: worker-deployment
+  triggers:
+    - type: kubernetes-resource
+      metadata:
+        resourceKind: "ConfigMap"
+        resourceName: "app-metrics"
+        key: "stats"
+        format: "yaml"
+        valueLocation: "application.workers.active"
+        valueType: "int64"
+        targetValue: "10"
+```
+
+In this example, the dot-separated path `application.workers.active` extracts the value `15` from the YAML data.
+
+#### Example 4: Quantity Type
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: resource-limits
+  namespace: keda
+data:
+  memory: |
+    {
+      "available": "2Gi",
+      "used": "512Mi"
+    }
+---
+apiVersion: keda.sh/v1alpha1
+kind: ScaledObject
+metadata:
+  name: memory-scaler
+  namespace: keda
+spec:
+  scaleTargetRef:
+    name: worker-deployment
+  triggers:
+    - type: kubernetes-resource
+      metadata:
+        resourceKind: "ConfigMap"
+        resourceName: "resource-limits"
+        key: "memory"
+        format: "json"
+        valueLocation: "used"
+        valueType: "quantity"
+        targetValue: "1024000000"  # 1Gi in bytes
+```
+
+In this example, the quantity string `"512Mi"` is parsed and converted to its float64 representation for scaling decisions.
+
+> ⚠️ **Important:** When using `valueType: "int64"` with JSON format, decimal values will be truncated. For example, if the JSON contains `{"count": 25.7}`, it will be treated as `25`.
+
+### Supported Formats and Value Extraction
+
+The scaler works in two steps:
+
+1. **Retrieve the raw string**: Read the value from `ConfigMap.data[key]` or `Secret.data[key]`
+2. **Extract the metric**: Parse the raw string based on the `format`:
+   - **number**: The raw string is parsed directly as a number. No `valueLocation` is needed.
+     - Example: If `key="count"` and `ConfigMap.data.count = "42"`, the metric value is `42`
+   - **json**: The raw string is parsed as JSON, then `valueLocation` (using GJSON syntax) extracts the metric value.
+     - Example: If `key="metrics"` and `ConfigMap.data.metrics = '{"tasks": 10}'`, with `valueLocation="tasks"`, the metric value is `10`
+   - **yaml**: The raw string is parsed as YAML, then `valueLocation` (using dot-separated path) extracts the metric value.
+     - Example: If `key="stats"` and `ConfigMap.data.stats = 'count: 15'`, with `valueLocation="count"`, the metric value is `15`
+
+### Limitations
+
+- Only resources in the same namespace as the ScaledObject are supported.
+- No cross-namespace access.
+- Only ConfigMaps and Secrets are supported.
+- The scaler uses the KEDA operator's service account permissions to access resources.
```

Comment:
might be worth mentioning how to configure RBAC for `Secrets` in the [helm chart](https://github.com/kedacore/charts/blob/5572cca92890c8bdd42ce190ceec929d5a0012f3/keda/templates/manager/clusterrole.yaml#L50)? Afaict, that one is not enabled by default as opposed to `ConfigMaps` that are [enbaled by default](https://github.com/kedacore/charts/blob/5572cca92890c8bdd42ce190ceec929d5a0012f3/keda/templates/manager/clusterrole.yaml#L16-L26).